### ISSUE TITLE
Python3 Support

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#227]
(Original issue 448 created by ashcrow on 2009-11-12T22:02:28+00:00)

I did a quick search looking for this projects plans for python3 use and didn't find any information. Is there a plan to have a separate branch to release tarballs from or will there be a specific cut off date where pygments moves to python3 (and users should use 2to3 for now)?

Either way, it would be nice to see this in the FAQ or another location on the wiki.

Reference: https://bugzilla.redhat.com/show_bug.cgi?id=537244

Thanks!

**Comment by Anteru** [comment]
(Original issue was assigned to gbrandl)

(Original issue 448 last updated on 2014-07-24T23:14:34.306190+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by gbrandl on 2009-11-16T17:09:06+00:00)

Pygments is already ported to Python 3.  The same source release can be used for 2.x and 3.x installs.  I'll add a note in the FAQ.

**Comment by Anteru** [comment]
(Original comment by matt_hulse on 2013-10-25T18:47:10.555036+00:00)

The FAQ says that Python 3 is supported but I receive the following error when attempting to import pygments in Python 3:


```
#!python

File "/home/user/src/pygments/__init__.py", line 46
    except TypeError, err:
```

From what I can tell, this should be:

```
#!python

    except TypeError as err:
```

The same code occurs on line 70.

**Comment by Anteru** [comment]
(Original comment by matt_hulse on 2013-10-25T19:31:34.036645+00:00)

I am running into several other issues in pygments as I am attempting to port a plugin that uses pygments from Python2 to Python3.

Am I supposed to be using 2to3 or something like that?

**Comment by Anteru** [comment]
(Original comment by andy9m on 2014-03-19T18:20:38.179901+00:00)

This should be reopened, as the problem mentioned by matt_hulse in October of last year is still preventing use of this library with python 3.

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2014-03-19T19:30:45.200086+00:00)

@Andy: All released Pygments versions are Python 3 compatible with 2to3, which is automatically called by setup.py.

Pygments 2.0, the current development version, will be single-source compatible (no 2to3) with 2.6+ and 3.3+.

**Comment by Anteru** [comment]
(Original comment by andy9m on 2014-03-20T18:36:12.970084+00:00)

Thanks Georg. I don't know why, but on my Mac OS X Mavericks system with python3 installed via homebrew, this wasn't the case. After your comment I tried

    pip3 install --force-reinstall -U pygments

and all was well.

**Comment by Anteru** [comment]
(Original comment by msabramo on 2014-07-24T23:14:34.291004+00:00)

So I was also getting this error when trying to pip install Pygments 1.6 into a python 3.3 virtualenv:

```
  File "/private/tmp/pygments-py33.venv/build/Pygments/pygments/__init__.py", line 46
    except TypeError, err:
                    ^
SyntaxError: invalid syntax
```

I scratched my head for a while wondering why 2to3 wasn't getting called.

Eventually I realized by doing `pip install -v Pygments` that pip was picking up a wheel that I had built. Something seems to be wrong with that wheel because after removing it, `pip install Pygments` works fine.

I mention this just in case anyone runs into this problem and goes searching and comes upon this issue.
